src/components/NavBar/UkBanner.tsx

```diff
@@ -0,0 +1,82 @@
+import { t, Trans } from '@lingui/macro'
+import { useOpenModal } from 'state/application/hooks'
+import { ApplicationModal } from 'state/application/reducer'
+import styled from 'styled-components'
+import { ButtonText, ThemedText } from 'theme/components'
+import { Z_INDEX } from 'theme/zIndex'
+
+export const UK_BANNER_HEIGHT = 64
+export const UK_BANNER_HEIGHT_MD = 112
+export const UK_BANNER_HEIGHT_SM = 136
+
+const BannerWrapper = styled.div`
+  position: relative;
+  display: flex;
+  background-color: ${({ theme }) => theme.surface1};
+  padding: 20px;
+  border-bottom: 1px solid ${({ theme }) => theme.surface3};
+  z-index: ${Z_INDEX.fixed};
+  box-sizing: border-box;
+  -webkit-box-sizing: border-box;
+  -moz-box-sizing: border-box;
+
+  @media only screen and (max-width: ${({ theme }) => `${theme.breakpoint.md}px`}) {
+    flex-direction: column;
+  }
+`
+
+const BannerTextWrapper = styled(ThemedText.BodySecondary)`
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+
+  @media only screen and (max-width: ${({ theme }) => `${theme.breakpoint.md}px`}) {
+    @supports (-webkit-line-clamp: 2) {
+      white-space: initial;
+      display: -webkit-box;
+      -webkit-line-clamp: 2;
+      -webkit-box-orient: vertical;
+    }
+  }
+
+  @media only screen and (max-width: ${({ theme }) => `${theme.breakpoint.sm}px`}) {
+    @supports (-webkit-line-clamp: 3) {
+      white-space: initial;
+      display: -webkit-box;
+      -webkit-line-clamp: 3;
+      -webkit-box-orient: vertical;
+    }
+  }
+`
+
+const ReadMoreWrapper = styled(ButtonText)`
+  flex-shrink: 0;
+  width: max-content;
+
+  :focus {
+    text-decoration: none;
+  }
+`
+
+export const bannerText = t`
+  UK disclaimer: This web application is provided as a tool for users to interact with the Uniswap Protocol on
```

Comment:
non-blocking nit: you get a double up of the `UK disclaimer` on the modal
![Screenshot 2023-10-06 at 9 04 40 AM](https://github.com/Uniswap/interface/assets/11512321/6df1ae05-421c-4202-b336-c790abb15f44)